### PR TITLE
add cd/activate cmds to explorer and file

### PR DESCRIPTION
**Pull request by pfitzseb**
Fixes https://github.com/julia-vscode/julia-vscode/issues/1742.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -145,7 +145,8 @@
             },
             {
                 "command": "language-julia.executeFile",
-                "title": "Julia: Execute File"
+                "title": "Julia: Execute File",
+                "icon": "$(play)"
```
Comment (pfitzseb) [inline package.json:149]: Unrelated fix to make sure we display an icon above the editor to run a file instead of the command's title.

File: src/interactive/repl.ts [hunk 2]
```diff
@@ -623,6 +658,8 @@ export function activate(context: vscode.ExtensionContext) {
         vscode.commands.registerCommand('language-julia.executeFile', executeFile),
         vscode.commands.registerCommand('language-julia.interrupt', interrupt),
         vscode.commands.registerCommand('language-julia.executeJuliaCodeInREPL', executeSelectionCopyPaste), // copy-paste selection into REPL. doesn't require LS to be started
+        vscode.commands.registerCommand('language-julia.cdHere', cdToHere), // copy-paste selection into REPL. doesn't require LS to be started
```
Comment (aviatesk) [inline src/interactive/repl.ts:661]:
```suggestion
        vscode.commands.registerCommand('language-julia.cdHere', cdToHere),
```

File: src/interactive/repl.ts [hunk 3]
```diff
@@ -623,6 +658,8 @@ export function activate(context: vscode.ExtensionContext) {
...
+        vscode.commands.registerCommand('language-julia.activateHere', activateHere), // copy-paste selection into REPL. doesn't require LS to be started
```
Comment (aviatesk) [inline src/interactive/repl.ts:662]:
```suggestion
        vscode.commands.registerCommand('language-julia.activateHere', activateHere),
```

File: src/interactive/repl.ts [hunk 4]
```diff
@@ -549,6 +550,40 @@ function interrupt() {
...
+    let uriPath = uri.fsPath
+    const stat = await fs.stat(uriPath)
+    if (stat.isFile()) {
+        return path.dirname(uriPath)
+    } else if (stat.isDirectory()) {
+        return uriPath
+    } else {
+        return null
+    }
+
```
Comment (aviatesk) [inline src/interactive/repl.ts:584]:
```suggestion
        return undefined
    }
```

File: src/interactive/repl.ts [hunk 5]
```diff
@@ -549,6 +550,40 @@ function interrupt() {
...
+        switchEnvToPath(uriPath, true)
+    }
+}
+
+async function getDirUriFsPath(uri: vscode.Uri) {
+    if (!uri) {
+        const ed = vscode.window.activeTextEditor
+        if (ed && ed.document && ed.document.uri) {
+            uri = ed.document.uri
+        }
```
Comment (aviatesk) [inline src/interactive/repl.ts:573]:
```suggestion
        }
        if (!uri) return undefined // bail out if there is no uri to retrieve
```

File: src/interactive/repl.ts [hunk 6]
```diff
@@ -549,6 +550,40 @@ function interrupt() {
...
+
+async function activateHere(uri: vscode.Uri) {
+    const uriPath = await getDirUriFsPath(uri)
+    if (uriPath) {
+        g_connection.sendNotification('repl/activateProject', uriPath)
+        switchEnvToPath(uriPath, true)
+    }
+}
+
+async function getDirUriFsPath(uri: vscode.Uri) {
```
Comment (aviatesk) [inline src/interactive/repl.ts:568]:
```suggestion
async function getDirUriFsPath(uri: vscode.Uri | undefined) {
```

for preparation for future strict mode ?

**Review by aviatesk — COMMENTED** [review]
Left few minor comments, but the code looks good to me in general.

Just to make sure, the current command definitions allows us to invoke them from explorer via GUI and also from keyboard shortcuts ? I would like to be able to invoke them from keyboard shortcut as in Juno.

File: package.json [hunk 7]
```diff
@@ -240,6 +241,14 @@
                 "command": "language-julia.debugEditorContents",
                 "title": "Julia: Debug File in New Process",
                 "icon": "$(debug-alt)"
+            },
+            {
+                "command": "language-julia.cdHere",
+                "title": "Julia: Change to this directory"
```
Comment (aviatesk) [inline package.json:247]:
```suggestion
                "title": "Julia: Change to This Directory"
```

File: package.json [hunk 8]
```diff
@@ -240,6 +241,14 @@
...
+            },
+            {
+                "command": "language-julia.activateHere",
+                "title": "Julia: Activate this environment"
```
Comment (aviatesk) [inline package.json:251]:
```suggestion
                "title": "Julia: Activate This Environment"
```

**Comment by pfitzseb** [comment]
> Just to make sure, the current command definitions allows us to invoke them from explorer via GUI and also from keyboard shortcuts ? I would like to be able to invoke them from keyboard shortcut as in Juno.

Yep. VSCode gives us the right URI in all contexts I've tried.

**Review by aviatesk — APPROVED** [review]
